### PR TITLE
feat(router-registry-service): add new interface for router registry

### PR DESCRIPTION
**Pull request by civing**
fix Lundalogik/crm-feature#2690

### CODE REVIEW

File: packages/interfaces/src/service/route-service.interface.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ import { Expression } from '../query.interface';
 
 /**
  * Service for routing
+ * @deprecated Has been replaced with a new routing service
```
Comment (jgroth) [inline packages/interfaces/src/service/route-service.interface.ts:5]:
```suggestion
 * @deprecated Has been replaced with the new `Navigator` service
```

**Review by jgroth — COMMENTED** [review]
Looks good to me, but as I said on the other PR we need to improve the documentation or create a new issue to fix that :slightly_smiling_face:

File: packages/interfaces/src/service/route-registry-service.interface.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+     */
+    registerRoute(urlPattern: string, component: string): void;
+
+    /**
+     * Find the component for a given URL.
+     *
+     * @param {string} url - The URL to match against.
+     * @return {MatchedComponent | undefined} Returns the component along with route parameters if found.
+     */
+    findComponent(url: string): MatchedComponent | undefined;
```
Comment (jgroth) [inline packages/interfaces/src/service/route-registry-service.interface.ts:22]: Should this be a URL or just the pathname? Did we decide to just use the pathname?

File: packages/interfaces/src/service/route-registry-service.interface.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ */
+export interface RouteRegistry {
+    /**
+     * Register a route for a component.
+     *
+     * @param {string} urlPattern - URL pattern used to match a component.
+     * For more information about URL patterns read [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/API/URL_Pattern_API).
+     * @param {string} component - The component to register to the given URL pattern.
+     */
+    registerRoute(urlPattern: string, component: string): void;
```
Comment (jgroth) [inline packages/interfaces/src/service/route-registry-service.interface.ts:14]: Kind of the same comment here, if we only require the pathname and if we recommend using that, maybe we should name this without including "URL" in the name

File: packages/interfaces/src/service/route-registry-service.interface.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    /**
+     * Matched path parameters.
+     */
+    props: Record<string, string>;
+
+    /**
+     * The extracted query parameters.
+     */
+    query: Record<string, string>;
```
Comment (jgroth) [inline packages/interfaces/src/service/route-registry-service.interface.ts:39]: This should probably be `Record<string, unknown>` if we are going to try and parse the values

File: packages/interfaces/src/service/route-registry-service.interface.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+export type MatchedComponent = {
+    /**
+     * The name of the component.
+     */
+    name: string;
+
+    /**
+     * Matched path parameters.
+     */
+    props: Record<string, string>;
```
Comment (jgroth) [inline packages/interfaces/src/service/route-registry-service.interface.ts:34]: Should we try and parse these as well? If so, we could change this to `Record<string, string | number>` or whatever makes more sense

**Review by jgroth — APPROVED** [review]
Looks good! Most of the comments are about improving the documentation and I think we can fix that in a separate issue so we can get this merged :slightly_smiling_face:

File: packages/interfaces/src/service/route-registry-service.interface.ts (hunk 5 above)
Comment (civing) [reply packages/interfaces/src/service/route-registry-service.interface.ts:34]: It makes sense to have them returned as numbers if they are numerics. 👍 We will fix this

File: packages/interfaces/src/service/route-registry-service.interface.ts (hunk 4 above)
Comment (civing) [reply packages/interfaces/src/service/route-registry-service.interface.ts:39]: Parse the values with JSON and treat them as unknown.

Comment (civing) [reply packages/interfaces/src/service/route-registry-service.interface.ts:39]: (We will move the search params and put it in the Location instead)

File: packages/interfaces/src/service/route-registry-service.interface.ts (hunk 5 above)
Comment (maltja) [reply packages/interfaces/src/service/route-registry-service.interface.ts:34]: fixed!

File: packages/interfaces/src/service/route-registry-service.interface.ts (hunk 3 above)
Comment (maltja) [reply packages/interfaces/src/service/route-registry-service.interface.ts:14]: ⚡ changed it to pathPattern instead

File: packages/interfaces/src/service/route-registry-service.interface.ts (hunk 2 above)
Comment (maltja) [reply packages/interfaces/src/service/route-registry-service.interface.ts:22]: ⚡  changed to path

File: packages/interfaces/src/service/route-registry-service.interface.ts (hunk 4 above)
Comment (maltja) [reply packages/interfaces/src/service/route-registry-service.interface.ts:39]: ⚡

**Review by jgroth — APPROVED** [review]

**Review by jgroth — APPROVED** [review]

**Pull request merged**